### PR TITLE
feat(wallet): implement search in saved addresses list

### LINKED ISSUE #20442: 🔍 Implement search in saved addresses list
# Feature Issue

### User Story

🔍 As a user, I want to search through the saved addresses list with name, ens, address and chain to find a saved address. 

### Description

Implement search in saved addresses list. The search can be performed with name, ens, address, and chain names. The search is local. Not require any RPC.

### Design 

[Link to Figma](https://www.figma.com/design/hLFPHYnuM0pOzgiabtqQ6k/Descoped-Wallet-settings-for-Mobile?node-id=74-33979&t=fjF8hC7hMdXV1ufw-1)

### Acceptance Criteria

| Screen | Description | Acceptance Criteria |
|--------|--------|--------|
| image | Display search input on top of the list |  Search input with placeholder `Name, ENS, or address` is displayed on top of the list |
| image | Allow searching with name, address, ENS and chain names |  1. The search can be performed with name or address or ENS or chain name <br/> 2. The list should reflect the search query <br/> | 
| image | Show nothing found illustration is shown if no matches |  An illustration with `Nothing found` is shown if there is no match or result |

### PR DESCRIPTION
**Pull request by smohamedjavid**
fixes #20442

### Summary

This PR adds a feature to search through the saved addresses list by name, address, ENS and chain names

https://github.com/status-im/status-mobile/assets/19339952/42802e9a-c30d-4dda-9208-331483ab7bea

### Testing notes

Testing will be done once the saved address EPIC is done

### Platforms

- Android
- iOS

##### Prerequisites: Have multiple saved addresses 

- Open Status
- Navigate to `Profile > Wallet > Saved addresses`
- Search with name and verify the saved address is filtered
- Search with ENS and verify the saved address is filtered
- Search with address and verify the saved address is filtered
- Search with chain short names and verify the saved address is filtered

status: ready

### CODE REVIEW

File: src/status_im/contexts/settings/wallet/saved_addresses/view.cljs [hunk 1]
```diff
@@ -50,15 +60,34 @@
...
+
+(defn- unfiltered-list
   []
-  [rn/view {:height 8}])
+  [rn/section-list
+   {:key-fn                          :title
+    :shows-vertical-scroll-indicator false
+    :sticky-section-headers-enabled  false
+    :render-section-header-fn        header
+    :sections                        (rf/sub [:wallet/grouped-saved-addresses])
```
Comment (mohsen-ghafouri) [inline src/status_im/contexts/settings/wallet/saved_addresses/view.cljs]: should it be called every time input is empty? or once at mount is enough?

File: src/status_im/contexts/settings/wallet/saved_addresses/view.cljs [hunk 2]
```diff
@@ -71,33 +100,44 @@
...
+                                                                       :on-press add-address-to-save}
+                                                 :blur?               true}
+
+                                          has-saved-addresses?
+                                          (assoc
+                                           :input       :search
+                                           :input-props {:placeholder         (i18n/label
+                                                                               :t/name-ens-or-address)
+                                                         :value               search-text
+                                                         :on-change-text      set-search-text
```
Comment (mohsen-ghafouri) [inline src/status_im/contexts/settings/wallet/saved_addresses/view.cljs]: maybe some debounce before call dispatch?

**Review by mohsen-ghafouri — APPROVED** [review]
LGTM, just some question/comment.

File: src/status_im/contexts/settings/wallet/saved_addresses/view.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/status_im/contexts/settings/wallet/saved_addresses/view.cljs]: At mount is enough. I need to move this to the main view. Fundamentally, subs are cached but it's best we move to the main view.

File: src/status_im/contexts/settings/wallet/saved_addresses/view.cljs (hunk 2 above)
Comment (smohamedjavid) [reply src/status_im/contexts/settings/wallet/saved_addresses/view.cljs]: Oops! I totally forgot it. NIce catch 👍

**Comment by smohamedjavid** [comment]
@status-im/mobile-qa - If it's not too much trouble, would you help check the failed test cases in the E2E results?

**Comment by churik** [comment]
e2e are not related to PR, thank you!

**Pull request merged**